x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/QuerySettings.java

```diff
@@ -53,7 +66,54 @@ public class QuerySettings {
         ZoneOffset.UTC
     );
 
-    public static final Map<String, QuerySettingDef<?>> SETTINGS_BY_NAME = Stream.of(PROJECT_ROUTING, TIME_ZONE)
+    @Param(name = "approximate", type = { "boolean", "map_param" }, description = "TODO - add description here")
+    @MapParam(
+        name = "approximate",
+        params = {
+            @MapParam.MapParamEntry(name = "num_rows", type = { "integer" }, description = "Number of rows."),
+            @MapParam.MapParamEntry(name = "confidence_level", type = { "double" }, description = "Confidence level.") }
+    )
+    public static final QuerySettingDef<Map<String, Object>> APPROXIMATE = new QuerySettingDef<>(
+        "approximate",
+        null,
+        false,
+        false,
+        true,
+        (value, ctx) -> {
+            Object res = null;
+            if (value instanceof Literal l) {
+                res = l.value();
+            } else if (value instanceof MapExpression me) {
+                try {
+                    res = me.toFoldedMap(FoldContext.small());
```

Comment:
Huh, `FoldContext`. Right now it's created by the `Configuration`, which depends on the `EsqlStatement` to be created, and therefore on the `QuerySettings`. Which is a _potentially problematic_ cyclical dependency.

The `FoldContext` is really created based on the pragmas though, there's no need for the Config. So this may be fine. We'll have to move its creation out of the Config at some point to make this work "correctly". But probably not worth it for this PR